packages/realm-web/src/WatchStream.ts

```diff
@@ -0,0 +1,270 @@
+
+        // ignore trailing CR from CRLF
+        if (line.endsWith("\r")) line = line.substr(0, line.length - 1);
+
+        if (line.length == 0) {
+            // This is the "dispatch the event" portion of the algorithm.
+            if (this._dataBuffer.length == 0) {
+                this._eventType = "";
+                return;
+            }
+
+            if (this._dataBuffer.endsWith("\n"))
+                this._dataBuffer = this._dataBuffer.substr(
+                    0,
+                    this._dataBuffer.length - 1,
+                );
+
+            this.feedSse({
+                data: this._dataBuffer,
+                eventType: this._eventType,
+            });
+            this._dataBuffer = "";
+            this._eventType = "";
+        }
+
+        if (line[0] == ":") return;
+
+        const colon = line.indexOf(":");
+        const field = line.substr(0, colon);
+        let value = colon == -1 ? "" : line.substr(colon + 1);
+        if (value.startsWith(" ")) value = value.substr(1);
+
+        if (field == "event") {
+            this._eventType = value;
+        } else if (field == "data") {
+            this._dataBuffer += value;
+            this._dataBuffer += "\n";
+        } else {
+            // line is ignored (even if field is id or retry).
+        }
+    }
+
+    feedSse(sse: ServerSentEvent) {
+        this.assertState(WatchStreamState.NEED_DATA);
+        const firstPercentIndex = sse.data.indexOf("%");
+        if (firstPercentIndex != -1) {
+            // For some reason, the stich server decided to add percent-encoding for '%', '\n', and '\r' to its
+            // event-stream replies. But it isn't real urlencoding, since most characters pass through, so we can't use
+            // uri_percent_decode() here.
+            let buffer = "";
+            let start = 0;
+            while (true) {
+                const percent =
+                    start == 0
+                        ? firstPercentIndex
+                        : sse.data.indexOf("%", start);
+                if (percent == -1) {
+                    buffer += sse.data.substr(start);
+                    break;
+                }
+
+                buffer += sse.data.substr(start, percent - start);
+
+                const encoded = sse.data.substr(percent, 3); // may be smaller than 3 if string ends with %
+                if (encoded == "%25") {
+                    buffer += "%";
+                } else if (encoded == "%0A") {
+                    buffer += "\x0A"; // '\n'
+                } else if (encoded == "%0D") {
+                    buffer += "\x0D"; // '\r'
+                } else {
+                    buffer += encoded; // propagate as-is
+                }
+                start = percent + encoded.length;
+            }
+
+            sse.data = buffer;
+        }
+
+        if (
+            sse.eventType === undefined ||
+            sse.eventType.length == 0 ||
+            sse.eventType == "message"
```

Comment:
Any reason not just to check by:
```ts
!sse.eventType || sse.eventType === "message"
```
... for example: would `null` imply a special meaning?

Reply:
No clue, but I made the change and the tests are green 🟢